src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,135 @@
+construction is parameterized by an ordinal `< ω₁`, and the cardinality bound is preserved along
+each step of the construction.
+-/
+
+universe u
+variables {α : Type u}
+
+open_locale cardinal
+open cardinal set
+
+local notation a `<₁` b := (aleph 1 : cardinal.{u}).ord.out.r a b
+local notation `ω₁`:= (aleph 1 : cardinal.{u}).ord.out.α
+
+namespace measurable_space
+
+/-- Transfinite induction construction of the sigma-algebra generated by a set of sets `s`. At each
+step, we add all elements of `s`, the empty set, the complements of already constructed sets, and
+countable unions of already constructed sets. We index this construction by an ordinal `< ω₁`, as
+this will be enough to generate all sets in the sigma-algebra. -/
+def generate_measurable_rec (s : set (set α)) : ω₁ → set (set α)
+| i := let S := ⋃ j : {j // j <₁ i}, generate_measurable_rec j.1 in
+    s ∪ {∅} ∪ compl '' S ∪ (set.range (λ (f : ℕ → S), ⋃ n, (f n).1))
+using_well_founded {dec_tac := `[exact j.2]}
+
+/-- At each step of the inductive construction, the cardinality bound `≤ 𝔠` is preserved. -/
+lemma cardinal_generate_measurable_rec_le_aux {s : set (set α)} (hs : #s ≤ 𝔠) (i : ω₁) :
+  #(generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  apply well_founded.induction (aleph 1 : cardinal.{u}).ord.out.wo.wf i,
+  assume i IH,
+  have J : #(⋃ (j : {j // j <₁ i}), generate_measurable_rec s j.1) ≤ 𝔠,
+  { apply (mk_Union_le _).trans,
+    have A : # {j // j <₁ i} ≤ aleph 1 := (mk_subtype_le _).trans (le_of_eq (aleph 1).mk_ord_out),
+    have B : cardinal.sup.{u u} (λ (j : {j // j <₁ i}), #(generate_measurable_rec s j.1)) ≤ 𝔠,
+    { apply cardinal.sup_le.2,
+      rintros ⟨j, hj⟩,
+      exact IH j hj },
+    apply le_trans (mul_le_mul' A B) _,
+    rw mul_eq_max (omega_le_aleph 1) omega_le_continuum,
+    apply max_le aleph_one_le_continuum le_rfl },
+  rw [generate_measurable_rec],
+  apply_rules [(mk_union_le _ _).trans, add_le_of_le omega_le_continuum, mk_image_le.trans,
+    mk_range_le.trans],
+  { rw [mk_singleton],
+    convert (nat_lt_continuum 1).le,
+    exact nat.cast_one.symm },
+  { simp only [mk_pi, subtype.val_eq_coe, prod_const, lift_uzero, mk_denumerable, lift_omega],
+    calc # (↥⋃ (j : {j // j <₁ i}), generate_measurable_rec s ↑j) ^ (ω : cardinal.{u})
+      ≤ 𝔠 ^ (ω : cardinal.{u}) : power_le_power_right J
+    ... = 𝔠 : continuum_power_omega }
+end
+
+lemma cardinal_generate_measurable_rec_le {s : set (set α)} (hs : #s ≤ 𝔠) :
+  #(⋃ i, generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  apply (mk_Union_le _).trans,
+  rw [(aleph 1).mk_ord_out],
+  exact le_trans (mul_le_mul' aleph_one_le_continuum
+    (cardinal.sup_le.2 (λ i, cardinal_generate_measurable_rec_le_aux hs i))) (by simp)
+end
+
+/-- A set in the sigma-algebra generated by a set of sets `s` is constructed at some step of the
+transfinite induction defining `generate_measurable_rec`. -/
+theorem generate_measurable_subset_rec (s : set (set α)) ⦃t : set α⦄
```

Comment:
I'm aware this isn't needed for this PR, but perhaps you could go one step further and prove that this is indeed the minimal σ-algebra?

Reply:
I'm not sure this will be ever useful, so I'd rather not devote time to it. I have added a comment about this in the docstring.